storm-core/src/clj/org/apache/storm/cluster.clj

```diff
@@ -123,6 +133,7 @@
 (def SUPERVISORS-SUBTREE (str "/" SUPERVISORS-ROOT))
 (def WORKERBEATS-SUBTREE (str "/" WORKERBEATS-ROOT))
 (def BACKPRESSURE-SUBTREE (str "/" BACKPRESSURE-ROOT))
+(def LEADERINFO-SUBTREE (str "/" LEADERINFO-ROOT))
```

Comment:
Is this needed?  we already have a way of getting the leader that is built into the client.  Why do we need a second way?

Reply:
Actually, this is not really needed, but if we always get leader from nimbus, we will make 2 RPC requests every time we contact with the leader [supervisor to nimbus/], instead get it from zookeeper directly is a better way.